apps/omg_child_chain/lib/omg_child_chain/release_tasks/set_fee_feed_adapter_opts.ex

```diff
@@ -0,0 +1,64 @@
+# Copyright 2019-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.ReleaseTasks.SetFeeFeedAdapterOpts do
+  @moduledoc false
+  use Distillery.Releases.Config.Provider
+  require Logger
+
+  @app :omg_child_chain
+  @config_key :fee_adapter_opts
+
+  @impl Provider
+  def init(_args) do
+    _ = Application.ensure_all_started(:logger)
+
+    @app
+    |> Application.get_env(:fee_adapter)
+    |> update_adapter_opts_with_env()
+  end
+
+  defp update_adapter_opts_with_env(nil), do: :ok
+
+  defp update_adapter_opts_with_env(OMG.ChildChain.Fees.FeedAdapter) do
+    adapter_opts =
+      @app
+      |> Application.fetch_env!(@config_key)
+      |> replace_with_env(&validate_string/2, feed_url: "FEE_FEED_URL")
```

Comment:
I'm not particularly fond of this approach - simply because it differs from the template of other release tasks. can we conform to the previous one - or adapt all of them to this?

Reply:
I'm happy you recognized my effort here. The reason it looks different is we operate in single config value which is Keyword.

I think I can export each `replace_with_env` to a separate function with a descriptive name so it benefit readability...

Reply:
can you show what kinda structure you're trying to build? that would be helpful!

Reply:
I've added the comment which explains how the adapter options are transformed